libfastpm/factors.c

```diff
@@ -197,28 +197,34 @@ fastpm_kick_store(FastPMKickFactor * kick,
     po->meta.a_v = af;
 }
 
-static double G_p(double a, FastPMCosmology * c)
+static double G_p(FastPMGrowthInfo * growth_info)
 {
     /* integral of G_p */
-    return GrowthFactor(a, c);
+    return growth_info->D1;
 }
-static double g_p(double a, FastPMCosmology * c)
+static double g_p(FastPMGrowthInfo * growth_info)
 {
-    return DGrowthFactorDa(a, c);
+    double dDda = growth_info->f1 * growth_info->D1 / growth_info->a;
+    return dDda;
 }
 
-static double G_f(double a, FastPMCosmology * c)           //CHECK gf Gf dont change with nus.
+static double G_f(FastPMGrowthInfo * growth_info)           //CHECK gf Gf dont change with nus.
 {
     /* integral of g_f */
-    return a * a * a * HubbleEa(a, c) * g_p(a, c);
+    double a = growth_info->a;
+    return a * a * a * HubbleEa(a, growth_info->c) * g_p(growth_info);
 }
 
-static double g_f(double a, FastPMCosmology * c)
+static double g_f(FastPMGrowthInfo * growth_info)
```

Comment:
The factors looked a bit different from those defined around / above https://github.com/rainwoodman/fastpm-python/blob/master/fastpm/background.py#L141 

The equations here seem to have assumed a matter only background. The factor of 3 indicates it is taking something like 'a^3' (the matter term). Perhaps also look into porting the more general equations over to here?

Reply:
Just checked, the factors are the same as the python version, just in python they've been written in terms of d/dloga instead of d/da. I don't think MD is assumed anywhere, but perhaps I'm wrong?

Reply:
Did you check the numerical values or the algebra?
I checked g_f and it looked about right.

I see you have also added radiation and nu to all orders if dE/da -- probably this is the only change that's needed?

Reply:
Just the algebra. Should I check the numerical values? What would I compare it to?